app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -996,134 +989,6 @@ class HomeFragment : Fragment() {
         requireComponents.core.tabCollectionStorage.register(collectionStorageObserver, this)
     }
 
-    /**
-     * This method will find and scroll to the row of the specified collection Id.
-     * */
-    private fun scrollAndAnimateCollection(
-        collectionIdToSelect: Long = -1
-    ) {
-        if (view != null && collectionIdToSelect >= 0) {
-            viewLifecycleOwner.lifecycleScope.launch {
-                val recyclerView = sessionControlView!!.view
-                delay(ANIM_SCROLL_DELAY)
-                val indexOfCollection =
-                    NON_COLLECTION_ITEM_NUM + findIndexOfSpecificCollection(collectionIdToSelect)
-
-                val lastVisiblePosition =
-                    (recyclerView.layoutManager as? LinearLayoutManager)?.findLastCompletelyVisibleItemPosition()
-                        ?: 0
-
-                if (lastVisiblePosition < indexOfCollection) {
-                    val onScrollListener = object : RecyclerView.OnScrollListener() {
-                        override fun onScrollStateChanged(
-                            recyclerView: RecyclerView,
-                            newState: Int
-                        ) {
-                            super.onScrollStateChanged(recyclerView, newState)
-                            if (newState == SCROLL_STATE_IDLE) {
-                                appBarLayout?.setExpanded(false)
-                                animateCollection(indexOfCollection)
-                                recyclerView.removeOnScrollListener(this)
-                            }
-                        }
-                    }
-                    recyclerView.addOnScrollListener(onScrollListener)
-                    recyclerView.smoothScrollToPosition(indexOfCollection)
-                } else {
-                    appBarLayout?.setExpanded(false)
-                    animateCollection(indexOfCollection)
-                }
-            }
-        }
-    }
-
-    /**
-     * Returns index of the collection with the specified id.
-     * */
-    private fun findIndexOfSpecificCollection(
-        changedCollectionId: Long
-    ): Int {
-        var result = 0
-        requireComponents.core.tabCollectionStorage.cachedTabCollections
-            .filterIndexed { index, tabCollection ->
-                if (tabCollection.id == changedCollectionId) {
-                    result = index
-                    return@filterIndexed true
-                }
-                false
-            }
-        return result
-    }
-
-    /**
-     * Will highlight the border of the collection with the specified index.
-     * */
-    private fun animateCollection(indexOfCollection: Int) {
```

Comment:
Did we discuss with Product or UX about whether or not they cared about keeping this interaction? Just want to know if we need to re-implement this or not.

Reply:
Removing this is a separate commit for which I've added in the description that a reassessment of the functionality is to happen in https://github.com/mozilla-mobile/fenix/issues/24389.